### DISCUSSION TITLE
Announcing Glow! — Beautiful, pixel-perfect Markdown code blocks

### DISCUSSION THREAD

**Discussion by tipiirai** [nuejs/nue#197] [Announcements]
<a href="https://nuejs.org/blog/introducing-glow/">
<img src="https://nuejs.org/img/glow-og-big.png"></a>
<br><br>

Check out the [announcement](https://nuejs.org/blog/introducing-glow/)

**Comment by libreom** [comment]
Make it responsive and take full scr een width in mobile
![Screenshot_20240213-163702_Brave.jpg](https://github.com/nuejs/nue/assets/111110599/c05a288f-daea-42a4-9465-0de9a6206411)

![Screenshot_20240213-163647_Brave.jpg](https://github.com/nuejs/nue/assets/111110599/84657dc4-a70b-4ee7-abc1-e103e361abcf)

**Reply by tipiirai** [reply]
Should be better now once your mobile browser cache is cleared (15 minutes): 
<img width="385" alt="image" src="https://github.com/nuejs/nue/assets/169330/0dd138dd-0315-4005-b5c9-04a50ef160f1">

**Reply by libreom** [reply]
Horizontal scrolling doesn't seem to work

**Reply by tipiirai** [reply]
On code blocks or the page?

**Reply by tipiirai** [reply]
Should be good now with a clean cache

**Reply by seneca** [reply]
Congrats on another milestone towards perfection. :-)

**Reply by libreom** [reply]
Works, I would prefer if you made codeblocks full width on mobile but that's your choice

**Comment by nobkd** [comment]
First, thank you for this incredible syntax highlighter!

---

I think I found a special case, where the highlighting does not work properly, because [python's floor division syntax (`//`)](https://docs.python.org/3/glossary.html#term-floor-division) looks like a line comment for other languages and therefore defaults to comment highlights:

```py
hello = 'world'

x = 5 + 3
y = 5 // 3
z = 5 % 3

# my comment
5 / 3  # my comment 2
5 // 3  # my comment 3
```

Rendered image:

![image](https://github.com/nuejs/nue/assets/44443899/d9dba3f0-f89b-4972-bbc8-ec42bcfcdbb7)

Rendered HTML:

```html
<pre glow=""><code language="py"><b>hello</b> <i>=</i> <em>'world'</em>

x <i>=</i> <em>5</em> <i>+</i> <em>3</em>
y <i>=</i> <em>5</em> <sup>// 3</sup>
z <i>=</i> <em>5</em> <i>%</i> <em>3</em>

<sup># my comment</sup>
<em>5</em> <i>/</i> <em>3</em>  <sup># my comment 2</sup>
<em>5</em> <sup>// 3  # my comment 3</sup></code></pre>
```

I haven't looked into the code for `glow` until now, so I'm not sure, how easy it is to fix this. :)

**Reply by tipiirai** [reply]
Thanks! Added [an issue](https://github.com/nuejs/nue/issues/212) for this one

**Reply by tipiirai** [reply]
Fixed it as well: https://github.com/nuejs/nue/commit/43302e22b486705ec9c3badd149763f0bb3816b0

**Comment by ArnavK-09** [comment]
Isnt this highlighting supposed to cover whole line? Two adjacent ones

![Screenshot_2024-02-15-11-20-42-60_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://github.com/nuejs/nue/assets/69188140/27d0b8a3-7d2d-4e62-93db-9a35ec1d4b67)

**Reply by tipiirai** [reply]
Thanks. Will fix

**Comment by ArnavK-09** [comment]
***Me waiting for __nuecss__**

***Tero with "Glow"**:-

![black-guy-in-suit-black-guy-suit.gif](https://github.com/nuejs/nue/assets/69188140/bb029b83-fbda-46db-8be6-cbdbb05154e3)

**Reply by tipiirai** [reply]
Hahaha! I'll get there :) 

Nue CSS is a rather big project. I'll push it out in pieces: updates to Nuekit and supportive documentation, like the Tailwind post last weekend.
